firmware/main/factory_reset.c

```diff
@@ -0,0 +1,179 @@
+						driver_framebuffer_getWidth(NULL)-driver_framebuffer_get_string_width(buffer, &permanentmarker_36pt7b),
+						driver_framebuffer_getHeight(NULL)-driver_framebuffer_get_string_height(buffer, &permanentmarker_36pt7b),
+						1,
+						1,
+						COLOR_BLACK,
+						&permanentmarker_36pt7b
+					);
+				}
+				driver_framebuffer_print(NULL, "BADGE.TEAM ESP32 PLATFORM", 0, driver_framebuffer_getHeight(NULL)-15, 1, 1, COLOR_BLACK, &fairlight_8pt7b);
+				driver_framebuffer_flush(force ? FB_FLAG_FORCE+FB_FLAG_FULL+FB_FLAG_LUT_NORMAL : FB_FLAG_LUT_FAST);
+			#endif
+			#if defined(CONFIG_DRIVER_SSD1306_ENABLE) || defined(CONFIG_DRIVER_ERC12864_ENABLE)
+				driver_framebuffer_fill(NULL, COLOR_FILL_DEFAULT);
+				driver_framebuffer_print(NULL, "Erasing storage...", 0, 0, 1, 1, COLOR_TEXT_DEFAULT, &roboto_12pt7b);
+				driver_framebuffer_print(NULL, text, 0, 14, 1, 1, COLOR_TEXT_DEFAULT, &roboto_12pt7b);
+				char buffer[16];
+				snprintf(buffer, 16, "%*u%%", 3, percentage);
+				driver_framebuffer_print(NULL, buffer, 0,30, 1, 1, COLOR_FILL_DEFAULT, &roboto_12pt7b);
+				uint16_t progressPosition = (percentage*(FB_WIDTH-17))/100;
+				driver_framebuffer_rect(NULL, 5, FB_HEIGHT-15, FB_WIDTH-10, 10, false, COLOR_TEXT_DEFAULT); //Outline of progress bar
+				driver_framebuffer_rect(NULL, 6, FB_HEIGHT-14, progressPosition, 8, true, COLOR_TEXT_DEFAULT); //Progress bar
+				driver_framebuffer_flush(0);
+			#endif
+			#ifdef CONFIG_DRIVER_HUB75_ENABLE
+				compositor_disable(); //Don't use the compositor if we have the framebuffer driver
+				driver_framebuffer_fill(NULL, COLOR_BLACK);
+				uint16_t progressPosition = (percentage*FB_WIDTH)/100;
+				for (uint8_t i = 0; i < FB_WIDTH; i++) {
+					driver_framebuffer_line(NULL, i, FB_HEIGHT-1, i, FB_HEIGHT-1, (i < progressPosition) ? 0x008800 : 0x880000);
+				}
+				if (!showPercentage) {
+					driver_framebuffer_print(NULL, text, 0, 0, 1, 1, COLOR_WHITE, &ipane7x5);
+				} else {
+					char buff[4];
+					sprintf(buff, "%d%%", percentage);
+					driver_framebuffer_print(NULL, buff, 0, 0, 1, 1, COLOR_WHITE, &ipane7x5);
+				}
+				driver_framebuffer_flush(0);
+			#endif
+		#else
+			#ifdef CONFIG_DRIVER_HUB75_ENABLE
+				compositor_enable();
+				char buff[4];
+				sprintf(buff, "%d%%", percentage);
+				Color col = {.value=0xffffffff};
+				compositor_clear();
+				compositor_addText(buff, col, 6, 1);
+			#endif
+		#endif
+}
+
+static void __attribute__((noreturn)) task_fatal_error(void)
+{
+	ESP_LOGE(TAG, "Exiting task due to fatal error...");
+	graphics_show_fr("Failed!", 0, false, true);
+	vTaskDelay(5000 / portTICK_PERIOD_MS);
+	esp_restart();
+	(void)vTaskDelete(NULL);
+}
+
+static void factory_reset_task(void *pvParameter)
+{
+	esp_err_t err;
+
+	graphics_show_fr("Erase starts in 2 seconds!", 0, false, true);
+	vTaskDelay(2000 / portTICK_PERIOD_MS);
+
+	const char* label = NULL; // Our FAT partition has label "locfd", not setting this wipes all FAT partitions on the internal flash chip

```

Comment:
I propose we set this to locfd, such that when we add the "initial system files" partition we discussed before, it will not get wiped by a factory reset.